src/wp-includes/class-wp-block-type.php

```diff
@@ -353,6 +367,10 @@ public function __get( $name ) {
 	 *              or false otherwise.
 	 */
 	public function __isset( $name ) {
+		if ( 'variations' === $name && isset( $this->variation_callback ) ) {
```

Comment:
```suggestion
		if ( 'variations' === $name && ( isset( $this->variation_callback ) || isset( $this->variations ) ) ) {
```

I think this is needed, note it is a private property.